src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseSharedAccessProtocolHttpsOnlyTests.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class UseSharedAccessProtocolHttpsOnlyTests : DiagnosticAnalyzerTestBase
+    {
+        protected void VerifyCSharpWithDependencies(string source, params DiagnosticResult[] expected)
+        {
+            string microsoftWindowsAzureStorageCSharpSourceCode = @"
+using System;
+
+namespace Microsoft.WindowsAzure.Storage
+{
+    public class CloudStorageAccount
+    {
+        public string GetSharedAccessSignature (SharedAccessAccountPolicy policy)
+        {
+            return """";
+        }
+    }
+
+    public sealed class SharedAccessAccountPolicy
+    {
+    }
+
+    public class IPAddressOrRange
+    {
+    }
+
+    public enum SharedAccessProtocol
+    {
+        HttpsOnly = 1,
+        HttpsOrHttp = 2
+    }
+
+    namespace File
+    {
+        public class CloudFile
+        {
+            public string GetSharedAccessSignature (SharedAccessFilePolicy policy, string groupPolicyIdentifier)
+            {
+                return """";
+            }
+
+            public string GetSharedAccessSignature (SharedAccessFilePolicy policy, SharedAccessFileHeaders headers, string groupPolicyIdentifier, Nullable<SharedAccessProtocol> protocols, IPAddressOrRange ipAddressOrRange)
+            {
+                return """";
+            }
+
+            public string GetSharedAccessSignature (SharedAccessFilePolicy policy, SharedAccessFileHeaders headers, string groupPolicyIdentifier, int protocols, IPAddressOrRange ipAddressOrRange)
```

Comment:
>int protocols [](start = 146, length = 13)

Where did you find this signature from?

Reply:
https://docs.azure.cn/zh-cn/dotnet/api/microsoft.windowsazure.storage.file.cloudfile?view=azure-dotnet

Reply:
Oh, this is a fake api which could be added potentially, I added this one to test the case when the signature of method didn't match.

Reply:
Oh okay, I couldn't find it documented or in the binaries.  At least add a comment in the stub API code and I think we can merge this.